src/report_1.R

```diff
@@ -1,3 +1,169 @@
-
+library(tidyverse)
+library(lubridate)
+library(tidyquant)
+library(extrafont)
+library(beepr)
+library(ggthemes)
+source("src/process_counts.R")
+# loadfonts() # <- possibly not needed
 ### `font_import()` should be run initially to ensure required fonts are present
 
+month_start <- function(month) {
+  as.Date(paste0(month, "-01"))
+}
+
+month_end <- function(month) {
+  as.Date(paste0(month, "-01")) + months(1) - days(1)
+}
+
+date_between <- function(start, end) {
+  out <- numeric(length = length(start))
+  for(i in seq_along(start)) {
+    out[i] <- sample(seq(min(start[i], end[i]),
+                         max(start[i], end[i]),
+                         by = "day"), 1)
+  }
+  as.Date(out)
+}
+
+impute_birthday <- function(birth_month, min_start, max_cease) {
+  earliest_possible <- max(max_cease - days(floor(18 * 365.25)) + 1, month_start(birth_month))
+  latest_possible <- min(min_start, month_end(birth_month))
+  date_between(earliest_possible, latest_possible)
+}
+
+year_diff <- function(start, stop) {
+  as.numeric(difftime(stop, start, units = "days")) %/% 365.25
+}
+
+theme_mastodon <- theme(plot.title = element_text(
+  hjust = 0.5, size = 20,
+  margin = margin(0,0,15,0)),
+  plot.caption = element_text(hjust = 0.5),
+  axis.title = element_text(
+    hjust = 0.5, size = 16),
+  axis.text = element_text(
+    hjust = 0.5, size = 10),
+  axis.text.x = element_text(angle = -45),
+  axis.title.x = element_text(margin = margin(15,0,0,0)),
+  axis.title.y = element_text(margin = margin(0,10,0,0)),
+  plot.margin = margin(10,20,10,10),
+  panel.grid = element_line(color = "#eeeeee"),
+  panel.background = element_rect(fill = "white",
+                                  colour = "black"))
+
+project_from <- as.Date("2020-03-31")
+project_yrs <- 4
+train_from <- project_from - years(3)
+projection_end <- project_from + years(project_yrs)
+
+report_1 <- function(actual_episodes_file, projected_episodes_file = NULL, counts_file = NULL) {
+  
+  plot(0:10, type = "n", xaxt="n", yaxt="n", bty="n", xlab = "", ylab = "")
+  text(5, 10, "Report 1", cex=2.5)
+  text(5, 8, "2021-XX-XX", cex=2.5) ## TODO take date from config
+  text(5, 6, "\"LA\" CiC Projection", cex=2.5) ## TODO take LA name from config
+  
+  set.seed(5)
+  colours <- c("gray", "#778899", "gray", "black", "blue")
+  names(colours) <- c("95% Simulation Range", "Projected IQR", "Projected Median", "SSDA903", "MIS")
+  dates <- seq(as.Date("2016-01-01"), projection_end, by = "week") ## TODO take dates from config file
+  tableau_color_pal("Tableau 20")(20)
+  
+  actual_episodes <- read.csv(actual_episodes_file, header = TRUE, 
+                              stringsAsFactors = FALSE, na.strings ="NA") %>% 
+    mutate(report_date = ymd(report_date),
+           ceased = ymd(ceased))
+  end_date <- max(c(actual_episodes$ceased, actual_episodes$report_date), na.rm = TRUE)
```

Comment:
should we have blank lines between the statements, or is that not really an R thing? (I know clojure is really only 1 line between functions rather than a pythonic 2